src/index/CompressedRelation.h

```diff
@@ -162,6 +166,20 @@ struct CompressedBlockMetadata : CompressedBlockMetadataNoBlockIndex {
     str << "block index: " << blockMetadata.blockIndex_ << "\n";
     return str;
   }
+
+  // Return true if a sequence of `CompressedBlockMetadata` is sorted, and if
+  // all the triples that are the same when disregarding the graph are in the
+  // same block.
+  static bool checkInvariantsForSortedBlocks(const auto& sequenceOfBlocks) {
+    return ::ranges::all_of(
+        ::ranges::views::sliding(sequenceOfBlocks, 2),
+        [](const auto& adjacent) {
+          const auto& first = adjacent.front().lastTriple_;
+          const auto& second = adjacent.back().firstTriple_;
+          return (first < second) &&
+                 (first.tieWithoutGraph() != second.tieWithoutGraph());
+        });
+  }
```

Comment:
This check should receive a unit test. This would fix the coverage and also make sure that this check works as expected (the logic is non-trivial imo)